File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/ContractionRepository.java
package fr.frogdevelopment.pregnancycalendar.data;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

import static fr.frogdevelopment.pregnancycalendar.data.AppDatabase.databaseWriteExecutor;

public class ContractionRepository {

    private final ContractionDao contractionDao;

    private final LiveData<List<Contraction>> allContractions;

    public ContractionRepository(Application application) {
        AppDatabase db = AppDatabase.getDatabase(application);
        contractionDao = db.contractionDao();
        allContractions = contractionDao.getAll();
    }

    public LiveData<List<Contraction>> getAllContractions() {
        return allContractions;
    }

    public void insert(Contraction contraction) {
        databaseWriteExecutor.execute(() -> contractionDao.insert(contraction));
    }

    public void delete(Contraction contraction) {
        databaseWriteExecutor.execute(() -> contractionDao.delete(contraction));
    }

    public void deleteAll() {
        databaseWriteExecutor.execute(contractionDao::deleteAll);
    }
}

File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/ContractionRepository.kt
package fr.frogdevelopment.pregnancycalendar.data

import android.app.Application
import androidx.lifecycle.LiveData

class ContractionRepository(application: Application) {

    private val contractionDao: ContractionDao
    val allContractions: LiveData<List<Contraction>>

    init {
        val db = AppDatabase.getDatabase(application)
        contractionDao = db.contractionDao()
        allContractions = contractionDao.all()
    }

    suspend fun insert(contraction: Contraction?) {
        contractionDao.insert(contraction)
    }

    suspend fun delete(contraction: Contraction?) {
        contractionDao.delete(contraction)
    }

    suspend fun deleteAll() {
        contractionDao.deleteAll()
    }
}